tests/test_directives.py

```diff
@@ -67,20 +73,22 @@ def test_default(self):
         assert MEMORY.default == 4
         assert PROCESS_FRACTION.default == 1.
 
-    def test_invalid_values(self):
-        invalid_values = {'np': [-1, 1., 'foo', {}, None],
-                          'ngpu': [-1, 1., 'foo', {}, None],
-                          'nranks': [-1, 1., 'foo', {}, None],
-                          'omp_num_threads': [-1, 1., 'foo', {}, None],
-                          'executable': [1, {}, None],
-                          'walltime': [-1., 'foo', {}, None],
-                          'memory': [-1, 4.5, 'foo', {}, None],
+    def test_invalid_values(self, available_directives_list):
+        invalid_values = {'np': [-1, 'foo', {}, None],
+                          'ngpu': [-1, 'foo', {}, None],
+                          'nranks': [-1, 'foo', {}, None],
+                          'omp_num_threads': [-1, 'foo', {}, None],
+                          'walltime': ['foo', {}, None],
+                          'memory': [-1, 'foo', {}, None],
                           'processor_fraction': [-0.5, 2.5, 'foo', {}, None]
                           }
-        for dir_name in invalid_values:
-            for dir_value in invalid_values[dir_name]:
-                with pytest.raises(Exception):
-                    dir.validation(dir_value)
+
+        for directive in available_directives_list:
+            if directive.name == 'executable':
+                continue
```

Comment:
Why are we skipping `executable` now?

Reply:
`__str__` methods are there for almost all data structures.
`Executable` expect a string, if not found, then it tries to convert it into a string and becomes successful almost every time.
Hence I thought that it'd be best to skip it for now

Reply:
Leave a comment about why `executable` is skipped.